tls/s2n_connection.h

```diff
@@ -267,18 +264,13 @@ struct s2n_connection {
     uint8_t alert_in_data[S2N_ALERT_LENGTH];
     struct s2n_stuffer alert_in;
 
-    /* An alert may be partially written in the outbound
-     * direction, so we keep this as a small 2 byte queue.
-     *
-     * We keep separate queues for alerts generated by
-     * readers (a response to an alert from a peer) and writers (an
-     * intentional shutdown) so that the s2n reader and writer
-     * can be separate duplex I/O threads.
+    /* Both readers and writers can trigger alerts.
+     * We prioritize writer alerts over reader alerts.
      */
-    uint8_t reader_alert_out_data[S2N_ALERT_LENGTH];
-    uint8_t writer_alert_out_data[S2N_ALERT_LENGTH];
-    struct s2n_stuffer reader_alert_out;
-    struct s2n_stuffer writer_alert_out;
+    uint8_t writer_alert_out;
+    uint8_t reader_alert_out;
+    uint8_t reader_warning_out;
+    bool alert_sent;
```

Comment:
I opened an issue for why we shouldn't do that: https://github.com/aws/s2n-tls/issues/4026

It's arguable whether alert_sent falls into this category, but I think it might. Our documentation doesn't currently say it, but I think we have to support calling s2n_shutdown_send while s2n_recv is still active.